src/phasorpy/_utils.py

```diff
@@ -260,3 +261,58 @@ def circle_line_intersection(
             y + (-dd * dx - abs(dy) * rdd) / dr,
         ),
     )
+
+
+def project_phasor_to_line(
+    real: ArrayLike,
+    imag: ArrayLike,
+    real_components: ArrayLike,
+    imag_components: ArrayLike,
+    /,
+    *,
+    clip: bool = True,
+    axis: int = -1,
+) -> tuple[NDArray, NDArray]:
+    """Return projected phasor coordinates to the line that joins two phasors.
+
+    By default, the points are clipped to the line segment between components
+    and the axis into which project the phasor can also be selected.
+
+    >>> project_phasor_to_line(
+    ...     [0.6, 0.5, 0.4], [0.4, 0.3, 0.2], [0.2, 0.9], [0.4, 0.3]
+    ... )  # doctest: +NUMBER
+    (array([0.592, 0.508, 0.424]), array([0.344, 0.356, 0.368]))
+
+    """
+    real = numpy.copy(real)
+    imag = numpy.copy(imag)
+    real_components = numpy.asarray(real_components)
+    imag_components = numpy.asarray(imag_components)
+    if real_components.size != 2:
```

Comment:
This check was something I added before making some changes to the function and just realized I left it there. Still I don't know if I should remove it. The idea of this is to check that it has only two components. I don't know if broadcasting works correctly, although I added the axis argument for this purpose. I tested a couple of cases and arrays of different shapes and I think t works, but still. 

If that is the case, I just realized that the other check:
```
    if numpy.all(real_components == imag_components):
        raise ValueError('components must have different coordinates')
```
should be modified to compare components in the same axis instead. 
What do you think?

Reply:
I don't see the `axis` argument. 
I wasn't sure what the `all(real_components == imag_components)` check was about. Instead test if `total_distance_between_components` is close to 0`?

Reply:
I like this approach. I will give it a try and update both functions. I only added the `axis` argument to `project_phasor_to_line` because I found it a little bit more challenging for `two_fractions_from_phasor`, but I will try to implement it for both.